src/pages/Library/Library.js

```diff
@@ -0,0 +1,203 @@
+              <MdWatchLater className="icon__large" />
+              <div
+                style={{
+                  fontSize: "1.2rem",
+                  fontWeight: "bold",
+                  paddingLeft: "1rem",
+                }}>
+                Watch Later
+              </div>
+            </div>
+            <Link
+              to="/watchlater"
+              className={`link ${styles.hover}`}
+              style={{ color: "var(--rb-primary)" }}>
+              SEE ALL
+            </Link>
+          </div>
+          <div className={styles.watchlaterVideos}>
+            {watchLaterPreview.map(({ video }) => (
+              <VideoCard
+                key={video._id}
+                video={video}
+                onOptionClick={handleOptionClick}
+              />
+            ))}
+          </div>
+        </section>
+        <section className={styles.section}>
+          <div className={styles.sectionHeader}>
+            <div className={styles.sectionTitle}>
+              <AiFillLike className="icon__large" />
+              <div
+                style={{
+                  fontSize: "1.2rem",
+                  fontWeight: "bold",
+                  paddingLeft: "1rem",
+                }}>
+                Liked Videos
+              </div>
+            </div>
+            <Link
+              to="/liked"
+              className={`link ${styles.hover}`}
+              style={{ color: "var(--rb-primary)" }}>
+              SEE ALL
+            </Link>
+          </div>
+          <div className={styles.likedVideos}>
+            {likePreview.map(({ video }) => (
+              <VideoCard
+                key={video._id}
+                video={video}
+                onOptionClick={handleOptionClick}
+              />
+            ))}
+          </div>
+        </section>
+        <section className={styles.section}>
+          <div className={styles.sectionHeader}>
+            <div className={styles.sectionTitle}>
+              <RiPlayListFill className="icon__large" />
+              <div
+                style={{
+                  fontSize: "1.2rem",
+                  fontWeight: "bold",
+                  paddingLeft: "1rem",
+                }}>
+                Playlists
+              </div>
+            </div>
+            <Link
+              to="/playlists"
+              className={`link ${styles.hover}`}
+              style={{ color: "var(--rb-primary)" }}>
+              SEE ALL
+            </Link>
+          </div>
+          <div className={styles.playlistedVideosPreview}>
+            {playListPreview?.map(({ _id, title, items }) => (
+              <Link
+                key={_id}
+                to={`/playlists/${_id}`}
+                className={`link ${styles.hover}`}>
+                <img
+                  className="image"
+                  src={
+                    items.length > 0
+                      ? generateThumbnail(items[0].video._id)
+                      : ""
```

Comment:
img src can't be an empty string, maybe put some placeholder (No Thumbnail)  there